### PR TITLE
Adds an explainer document

### PR DESCRIPTION
**Pull request by danielkhan**

### CODE REVIEW

File: EXPLAINER.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+We propose a set of HTTP headers which propagate a distributed trace, even when a request and its associated trace is served by multiple tracing providers.
+
+## Why do we care?
+
+Because there is no standardized way to propagate a trace, each tracing vendor has to create their own method of propagating tracing information. Typically this is in the form of a custom HTTP header. If an application request which is traced by one system flows through a system which is not traced using the same tracing platform, the headers may not be correctly propagated and the trace may be broken.
+
+## Goals
+
+- To provide a way in which heterogeneous architectures with multiple tracing platforms to operate in concert with each other
```
Comment (subvocal) [inline EXPLAINER.md]: Suggest rewording to be more clear:
To provide a standard mechanism to propagate shared context across various architecture components so that multiple tracing platforms can be used in concert with each other.

File: EXPLAINER.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+The traceparent header is a request header which contains a trace id, a parent span id, and trace flags. A tracing system transmits the trace ID, span ID, and the ID of each span's parent span, if it exists, with each span to a tracing backend. The tracing backend uses this information to construct the directed acyclic graph which represents the trace. In a system with a single tracing platform, it is the only header necessary to complete a distributed trace.
+
+### Tracestate
+
+The tracestate header is a request header which allows a tracing platform to transmit platform-specific information, even through systems which are traced using other tracing platforms. Each tracing platform uses a single key, and the value is treated as an opaque token by other tracing platforms. Each tracing platform is required to forward tracestate keys and values.
+
+### Trace Response header
+
+We also propose a response header which can be used to report a trace ID back to the caller to
+- report back a new trace ID in case the callee, ignored the existing headers but started a new trace
```
Comment (subvocal) [inline EXPLAINER.md]: remove the comma in

> callee, ignored

File: EXPLAINER.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- let proxies delegate sampling decisions to the caller
+- allow tail-based sampling, where a sampling decision is deferred until a request is completed and all information pertinent to the sampling decision is known
+- to correlate the initial page load of a browser and all subsequent requests to a server side trace
+
+### Baggage
+
+The baggage header is used to propagate properties not defined in trace parent.
+There are two common use cases. First is to define a context on trace initiation.
```
Comment (subvocal) [inline EXPLAINER.md]: What's the second use case? Maybe put in a numbered list?

File: EXPLAINER.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+2. The tracing system uses a proprietary header to propagate its trace ID.
+3. The API gateway is not configured to propagate the proprietary header
+4. Service C does not receive a trace ID and context is lost
+
+![Context loss due to a middleware](./assets/explainer_context_loss.png "Context loss due to a middleware")
+
+### Context propagation with W3C Trace Context
+1. Service A calls Service B through an API gateway
+2. The tracing system uses the W3C Trace Context header
+3. The API gateway operates standard compliant
```
Comment (subvocal) [inline EXPLAINER.md]: Suggest rewording to:
The API gateway recognizes the headers as standards-compliant and allows them through

**Review by subvocal — CHANGES_REQUESTED** [review]
I really like the images as a way to explain what trace context is. Could you add the explainer_context_preserved.png image to the first paragraph that illustrates this scenario? Maybe make it clear in the image that Service A is instrumented by one tracing solution and Service B is instrumented by a different tracing solution?

File: EXPLAINER.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
+# Distributed Tracing Working Group
+
+## What is it?
+
+We propose a set of HTTP headers which propagate a distributed trace, even when a request and its associated trace is served by multiple tracing providers.
```
Comment (dyladan) [inline EXPLAINER.md]: Baggage does not really fit into this description.
```suggestion
We propose a set of HTTP headers which propagate a distributed trace and associated data, even when a request and its associated trace is served by multiple tracing providers.
```

File: EXPLAINER.md (hunk 3 above)
Comment (danielkhan) [reply EXPLAINER.md]: I changed the whole section to contain a list of known use cases instead of randomly picking two

File: EXPLAINER.md (hunk 4 above)
Comment (danielkhan) [reply EXPLAINER.md]: Done

**Review by yurishkuro — COMMENTED** [review]
Not really sure what the main objective of this document is, seems like a mix of different things.

**Comment by danielkhan** [comment]
> Not really sure what the main objective of this document is, seems like a mix of different things.

We were asked to provide an explainer that covers what the working group is doing. This is more-or-less a companion doc to the charter.

**Comment by danielkhan** [comment]
> I really like the images as a way to explain what trace context is. Could you add the explainer_context_preserved.png image to the first paragraph that illustrates this scenario? Maybe make it clear in the image that Service A is instrumented by one tracing solution and Service B is instrumented by a different tracing solution?

Added an end-to-end example image.

**Comment by danielkhan** [comment]
I've incorporated all the requested changes. @mtwo, @SergeyKanzhelev, @victorNewRelic, @dyladan, @yurishkuro PTAL

**Review by subvocal — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]
